User interface Milestone

AhmedAmin90 opened the pull request:
In this milestone, I created a mock for the game steps.
Specifically, I created steps that will collect input from the user and what will be displayed for the user and mock the basic flow of the game.

AhmedAmin90 committed:
Add linter files

AhmedAmin90 committed:
Add bin and lib file with their internal files

AhmedAmin90 committed:
Start in the mock

AhmedAmin90 committed:
Add commented steps for the game

AhmedAmin90 committed:
Add the mock the game

AhmedAmin90 committed:
Fix linters

AhmedAmin90 committed:
Add greeting msg in the first

AhmedAmin90 committed:
fix linters

AhmedAmin90 committed:
fix

kevcap reviewed (changes requested):
# [STATUS] CHANGES REQUIRED♻️

#### Hey team,  your project is looking really good, however, to accomplish as the actual project, you required to do these changes, listed below: 
 
## [REQUIRED CHANGES]

Correcton of some typos:
First player's name; (Sounds better than Name of the first player)
Name of the first player; (add **the**)
Name of the second player; (add **the** and remove the **E** at the end of Second**E**)
Select a number between 1 **and** 9; (instead of **to**)

- [x] When the user selects an occupated space, your message should be something like: That space is already taken, please select another one. Instead of: Invalid number.
- [x]  The game must inform a player if the selected move is a winning move (without computing the winning move)
- [x] The game must inform a player if the selected move is a draw move (without computing draw)

## Please make the required changes and resubmit for a new code review once you are ready💪

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> <img  alt="gmail"  src="https://img.shields.io/badge/gmail-%20- green?logo=gmail&style=social"></a> [![Twitter Follow](https://img.shields.io/twitter/follow/kevinnnmn?style=social)](https://twitter.com/intent/follow?screen_name=kevinnnmn) [![](https://img.shields.io/badge/Slack-Kevin%20Alves-green?logo=slack&style=social)](https://microverse-students.slack.com/team/UT86D3AP4)

<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In the real world, you'll have a lot of change requests for your real clients, it's good to get used to it.</sub>

AhmedAmin90 committed:
Modify condition , add win and draw (not computing) method

AhmedAmin90 committed:
Fix rubocop

AhmedAmin90 committed:
Merge branch 'development' into user-interface

AhmedAmin90 committed:
remove blank lines

descholar-ceo commented:
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

descholar-ceo commented on bin/main.rb:
```diff
@@ -1,3 +1,97 @@
+count = 0
+while flag == false
+
+  puts "#{player_one} turn: Select number between 1 To 9"
```
Use between `1 and 9`

descholar-ceo commented on bin/main.rb:
```diff
@@ -1,3 +1,97 @@
+  # Change the board if check pass : Now, the move is displayed on the board.
+  # Or: Write warning message if invalid number or repeated number.
+
+  puts "#{player_two} turn: Select number between 1 To 9"
```
Use between `1 and 9`

descholar-ceo reviewed (changes requested):
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @AhmedAmin90  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] You should fix the typos that you are told in the first review, please follow the suggestions the first reviewer pointed out
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

AhmedAmin90 committed:
Fix Typos

I fixed the first one in the method and I forget to fix the others in the rest of the code. I am falling in love with the next milestone (game logic) 😁 so that is why I forget to fix them. That is my fault 😇.

descholar-ceo commented:
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

descholar-ceo reviewed (changes requested):
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
## :clap: :clap: :clap: Congratulations @AhmedAmin90  , your project is approved :tada: :tada: :tada: 
You have implemented all minimum requirements for this project to mark it as complete, it doesn't need more changes, you can go ahead and merge all of the changes on base branch,

Don't forget to fill the mandatory project completion form, and see you to next project,

Happy coding :computer: :computer: :computer: 

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)
